### PR TITLE
fix(gnovm): handle type alias declaration for PrimitiveType

### PR DESCRIPTION
**Pull request by hthieu1110**
Fixes: https://github.com/gnolang/gno/issues/3203

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
</details>

### CODE REVIEW

**Review by mvertes — APPROVED** [review]
Looks good to me

File: gnovm/pkg/gnolang/preprocess.go [hunk 1]
```diff
@@ -2352,6 +2352,8 @@ func preprocess1(store Store, ctx BlockNode, n Node) Node {
 						// }
 						*dst = *dt2
 					}
+				case PrimitiveType:
```
Comment (omarsy) [inline gnovm/pkg/gnolang/preprocess.go:2355]: WDYT, handle other types like PointerType, NativeType ?
example this:

```
package main

type (
	Number  = *int32
	Number2 = Number
)

func f(n Number) { println(n) }

func main() {
	var n int32 = 5
	f(&n)
}
```

Comment (hthieu1110) [reply gnovm/pkg/gnolang/preprocess.go:2355]: I think it's good idea

Comment (hthieu1110) [reply gnovm/pkg/gnolang/preprocess.go:2355]: @omarsy just found this: https://github.com/gnolang/gno/issues/1361 which confirms that we dont need to process NativeType

**Review by petar-dambovaliev — APPROVED** [review]

**Pull request merged**